### ISSUE TITLE
Your operating system denied the spawn of C:/Users/feng/.cargo/bin/racer.exe process: environment can only contain strings

### ISSUE THREAD

**Issue by fengcen** [DamnWidget/anaconda_rust#9]
anaconda_rust won't auto complete, but create tmp files when i typing.
anaconda_jsonserver.log show errors below:

```
2016-06-30 21:47:31,930: INFO    : client requests: autocomplete
2016-06-30 21:47:31,934: ERROR   : Your operating system denied the spawn of C:/Users/feng/.cargo/bin/racer.exe process: environment can only contain strings
2016-06-30 21:47:31,934: ERROR   : The underlying racer tool raised an exception
2016-06-30 21:47:31,934: ERROR   : Your operating system denied the spawn of C:/Users/feng/.cargo/bin/racer.exe process: environment can only contain strings
2016-06-30 21:47:31,936: DEBUG   : Traceback (most recent call last):
File "D:\Greensoft\sublime-text-3\Data\Packages\Anaconda\anaconda_server\lib\../../../anaconda_rust/plugin\handlers_rust\commands\autocomplete.py", line 36, in run
  'completions': self.completions(),
File "D:\Greensoft\sublime-text-3\Data\Packages\Anaconda\anaconda_server\lib\../../../anaconda_rust/plugin\handlers_rust\commands\autocomplete.py", line 71, in completions
  proc = spawn(args, stdout=PIPE, stderr=PIPE, cwd=os.getcwd(), env=env)
File "D:\Greensoft\sublime-text-3\Data\Packages\Anaconda\anaconda_server\process.py", line 30, in spawn
raise RuntimeError(msg)
RuntimeError: Your operating system denied the spawn of C:/Users/feng/.cargo/bin/racer.exe process: environment can only contain strings
```

**Comment by fengcen** [comment]
I solve this probleam by remove "rust_src_path" in AnacondaRust Settings - User.
But there another probleam:

```
2016-06-30 22:26:26,069: INFO    : client requests: autocomplete
2016-06-30 22:26:26,125: ERROR   : The underlying racer tool raised an exception
2016-06-30 22:26:26,125: ERROR   : thread 'searcher' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore\option.rs:325
note: Run with `RUST_BACKTRACE=1` for a backtrace.
ERROR:racer: Search thread paniced: Any

2016-06-30 22:26:26,127: DEBUG   : Traceback (most recent call last):
File "D:\Greensoft\sublime-text-3\Data\Packages\Anaconda\anaconda_server\lib\../../../anaconda_rust/plugin\handlers_rust\commands\autocomplete.py", line 36, in run
'completions': self.completions(),
File "D:\Greensoft\sublime-text-3\Data\Packages\Anaconda\anaconda_server\lib\../../../anaconda_rust/plugin\handlers_rust\commands\autocomplete.py", line 82, in completions
raise Exception(err)
Exception: thread 'searcher' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore\option.rs:325
note: Run with `RUST_BACKTRACE=1` for a backtrace.
ERROR:racer: Search thread paniced: Any
```

**Comment by DamnWidget** [comment]
The first problem is because the `env` variable contains something that is not a string (that is weird), the second one is because you removed the `rust_src_path` from your config and `racer` is not able to find the Rust standard library code.

**Comment by fengcen** [comment]
The first problem, I found anaconda_rust convert `rust_src_path` to unicode object, same problem as this link: [http://stackoverflow.com/questions/12253014/why-does-popen-fail-on-windows-if-the-env-parameter-contains-a-unicode-object](http://stackoverflow.com/questions/12253014/why-does-popen-fail-on-windows-if-the-env-parameter-contains-a-unicode-object)
I remove `rust_src_path`, but I had set `RUST_SRC_PATH` env variable in windows, so the second problem may be not 

> racer is not able to find the Rust standard library code.

**Comment by fengcen** [comment]
I downgrade `racer` from `v1.2.10` to `v1.1.0`, the second problem gone... `v1.2.9` still has this problem.

**Comment by DamnWidget** [comment]
I will take a look on my windows box when I get home

**Comment by DamnWidget** [comment]
Fixed in [v0.2.1](https://github.com/DamnWidget/anaconda_rust/releases/tag/v0.2.1), feel free to open it again if you encounter some problem with the fix

**Comment by fengcen** [comment]
Thanks.
